aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -206,6 +213,11 @@ private boolean isUnauthorizedException(DataStoreException exception) {
         .subscribeOn(Schedulers.io())
         .observeOn(Schedulers.io())
         .map(SubscriptionProcessor::unwrapResponse)
+        .filter(modelWithMetadata -> {
+            Class<? extends Model> clazz = modelSchema.getModelClass();
```

Comment:
If you do the `String modelName` instead of the `Class<? extends Model> modelClass`, could be just `modelSchema.getName()`